src/ensembl/src/content/app/browser/Browser.test.tsx

```diff
@@ -1,5 +1,95 @@
-describe.skip('<Browser />', () => {
-  test('it works', () => {
-    // this is a reminder to test the Browser component
+import React from 'react';
+import { MemoryRouter } from 'react-router';
+import { Provider } from 'react-redux';
+import { mount } from 'enzyme';
+import faker from 'faker';
+
+import { Browser, BrowserProps, ExampleObjectLinks } from './Browser';
+import BrowserImage from './browser-image/BrowserImage';
+import TrackPanel from './track-panel/TrackPanel';
+import AppBar from 'src/shared/components/app-bar/AppBar';
+
+import configureStore from 'src/store';
+
+jest.mock('static/browser/browser.js', () => {});
+
+describe('<Browser />', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  const defaultProps: BrowserProps = {
+    activeGenomeId: faker.lorem.words(),
+    activeEnsObjectId: faker.lorem.words(),
+    allActiveEnsObjectIds: {
+      [faker.lorem.words()]: faker.lorem.words()
+    },
+    allChrLocations: { [faker.lorem.words()]: ['1', 10, 5000] },
+    browserActivated: false,
+    browserNavOpened: false,
+    browserQueryParams: {},
+    chrLocation: ['1', 10, 5000],
+    isDrawerOpened: false,
+    isTrackPanelOpened: false,
+    launchbarExpanded: false,
+    exampleEnsObjects: [],
+    committedSpecies: [],
+    changeBrowserLocation: jest.fn(),
+    changeFocusObject: jest.fn(),
+    changeDrawerView: jest.fn(),
+    closeDrawer: jest.fn(),
+    fetchGenomeData: jest.fn(),
+    replace: jest.fn(),
+    toggleDrawer: jest.fn(),
+    setDataFromUrlAndSave: jest.fn()
+  };
+
+  const store = configureStore();
+
+  const wrappingComponent = (props: any) => (
+    <MemoryRouter>
+      <Provider store={store}>{props.children}</Provider>
+    </MemoryRouter>
+  );
+
+  const mountBrowserComponent = (props?: Partial<BrowserProps>) =>
+    mount(<Browser {...defaultProps} {...props} />, { wrappingComponent });
+
+  describe('rendering', () => {
+    test('does not render when no activeGenomeId', () => {
+      const wrapper = mountBrowserComponent({ activeGenomeId: null });
+      expect(wrapper.find(AppBar)).toHaveLength(0);
+    });
+
+    test('renders links to example objects if there is no selected focus object', () => {
+      const wrapper = mountBrowserComponent();
+      expect(wrapper.find(ExampleObjectLinks)).toHaveLength(1);
+    });
+
+    test('renders the genome browser and track panel when if there is a selected focus object', () => {
+      const wrapper = mountBrowserComponent({
+        browserQueryParams: { focus: faker.lorem.words() }
+      });
+      expect(wrapper.find(BrowserImage)).toHaveLength(1);
+      expect(wrapper.find(TrackPanel)).toHaveLength(1);
+    });
+  });
```

Comment:
`when if` -> `only if`? 

Also, would it be better to test if the length is 0 when there is no focus?

Reply:
I didn't test that since we assume it is going to be the opposite when there is no selected focus object. However, I am curious if we should also test the opposite effects in our tests.

Reply:
If the `BrowserImage` should get rendered only if the focus is present, there is a chance that it could always get rendered without failing any tests.

Reply:
That sounds right. I'll add these assertions to the same test case and change the wording of the description.